extensions/wikia/FlowTracking/scripts/flowTrackingCreatePage.js

```diff
@@ -0,0 +1,49 @@
+define(
+	'wikia.flowTracking.createPage',
+	['wikia.flowTracking', 'wikia.querystring', 'mw', 'wikia.document', 'wikia.window'],
+	function (flowTrack, QueryString, mw, document, window) {
+		'use strict';
+
+		var namespaceId = mw.config.get('wgNamespaceNumber'),
+			articleId = mw.config.get('wgArticleId');
+
+		function trackOnEditPageLoad(editor) {
+			var qs = new QueryString(),
+				// 'flow' is the parameter passed in the url if user has started a flow already
+				flowParam = qs.getVal('flow', false),
+				tracked = qs.getVal('tracked', false);
+
+			// Do not track if the step was tracked already
+			if (tracked) {
+				return;
+			}
+
+			// Track only creating articles (wgArticleId=0) from namespace 0 (Main)
+			// IMPORTANT: on Special:CreatePage even after providing article title the namespace is set to -1 (Special Page)
+			if (namespaceId === 0 && articleId === 0) {
+				if (flowParam || document.referrer) {
+					flowTrack.trackFlowStep(flowParam, {editor: editor});
+				} else {
+					flowTrack.beginFlow(window.wgFlowTrackingFlows.CREATE_PAGE_DIRECT_URL, {editor: editor});
+					qs.setVal('flow', window.wgFlowTrackingFlows.CREATE_PAGE_DIRECT_URL);
+				}
+			}
+
+			// For Special:CreatePage
+			if (namespaceId === -1 && articleId === 0) {
```

Comment:
This and previous blocks could we unified as `trackFlowStep` is done the same way for both.